### PR TITLE
Process transactions and solutions in parallel

### PR DESCRIPTION
**Pull request by vicsn**
## Motivation

As suggested by @ljedrz in [this comment](https://github.com/AleoNet/snarkOS/pull/3421/files#r1837739260), processes transactions and solutions in parallel. This is a small cleanup and speedup.

## Test Plan

- [x] Ran a local network

### CODE REVIEW

File: node/consensus/src/lib.rs [hunk 1]
```diff
@@ -475,13 +475,11 @@ impl<N: Network> Consensus<N> {
...
-                    warn!("Cannot process unconfirmed transactions - {e}");
-                }
-                // Process the unconfirmed solutions in the memory pool.
-                if let Err(e) = self_.process_unconfirmed_solutions().await {
-                    warn!("Cannot process unconfirmed solutions - {e}");
+                // Process the unconfirmed transactions and solutions in the memory pool.
+                if let Err(e) =
+                    tokio::try_join!(self_.process_unconfirmed_transactions(), self_.process_unconfirmed_solutions(),)
+                {
+                    warn!("Cannot process unconfirmed transactions and solutions - {e}");
```
Comment (ljedrz) [inline node/consensus/src/lib.rs]:
```suggestion
                    warn!("Cannot process unconfirmed transactions or solutions - {e}");
```
nit

**Review by ljedrz — APPROVED** [review]
LGTM; also, a tiny technical note - this _may_ cause these operations to be performed in parallel, but they are bound to be run concurrently.

File: node/consensus/src/lib.rs (hunk 1 above)
Comment (vicsn) [reply node/consensus/src/lib.rs]: Thx, I went back and forth on this one 😅

**Review by Pjrich1313 — APPROVED** [review]

**Review by zkxuerb — APPROVED** [review]

**Comment by raychu86** [comment]
I think this will still run them concurrently unless the underlying function has it's own thread: https://dtantsur.github.io/rust-openstack/tokio/macro.try_join.html.

Also, I think this will lead to some unwanted consequences - if the first call fails, it will not run the other one.

**Comment by vicsn** [comment]
> Also, I think this will lead to some unwanted consequences - if the first call fails, it will not run the other one.

Good catch that there's an early return on `Err`, don't think any potential speedup will be worth the risk - at this stage.

**Pull request closed without merging**